qiskit_machine_learning/neural_networks/opflow_qnn.py

```diff
@@ -80,6 +80,16 @@ def __init__(self, operator: OperatorBase,
         super().__init__(len(self._input_params), len(self._weight_params),
                          sparse=False, output_shape=output_shape)
 
+    @property
```

Comment:
I think the comments here are similar. The setter should allow Backend... and wrap it like in the constructor. Also I think it again needs to take care of CircuitSampler, that is setup in the constructor if the quantum instance is updated we need to redo that I think off the new quantum instance